src/main/java/com/github/leroyguillaume/keycloak/bcrypt/BCryptPasswordHashProvider.java

```diff
@@ -51,13 +55,13 @@ public void close() {
     public boolean verify(final String rawPassword, final PasswordCredentialModel credential) {
         final String hash = credential.getPasswordSecretData().getValue();
         char[] pw = rawPassword.toCharArray();
-        final BCrypt.Result longPasswordVerifier = BCrypt.verifyer(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR, LongPasswordStrategies.truncate(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR)).verify(pw, hash);
-        if (longPasswordVerifier.verified) {
-            return longPasswordVerifier.verified;
+        try {
+            final BCrypt.Result verifier = BCrypt.verifyer(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR, LongPasswordStrategies.truncate(BCrypt.Version.VERSION_2Y_NO_NULL_TERMINATOR)).verify(pw, hash);
```

Comment:
Should we check why truncate is not a fix for this? that's the purpose of it no?

Additionally we can't do the the fallback just in the case of an exception. We have to attempt it even if there's no exception when verify fails.